typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,125 @@
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: bindingsJsGrpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, [], Error] { //TODO - [int32, pointer to []byte (comment), error ] ; is is not []`type` like []string
+        let serviceId = this.serviceId;
+        let args = constructors.newGetExecCommandArgs(serviceId, command);
+        let request: grpc.requestCallback<bindingsJs.ExecCommandResponse>;
+
+        let resp, err = this.client.execCommand(args, request); //TODO - does error checking work here ? 
+        
+        if (err != null) {
+            return [0, null, new Error( //TODO - Error type (do I need callback?)
+			    "An error occurred executing command '%v' on service '%v'", //TODO - I might not be able keep the '%v'
+            )]; //TODO (comment) cannot return multipe values in typescript function, need to wrap in array or object
+        }
+        return resp.ExitCode, resp.LogOutput, null; //TODO (comment) resp.LogOuput was a pointer
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public generateFiles(filesToGenerateSet: Set<string>): [Map<string, GeneratedFileFilepaths>, Error] {
+        let serviceId = this.serviceId;
+        let fileGenerationOpts = new Map(); //TODO - Still need map since both key and value are useful ; Key (Set just needs key): string & Value: core_api_bindings.FileGenerationOptions{} - type not needed in typescript
+        for (let fileId in filesToGenerateSet) {
+            fileGenerationOpts[fileId] = new bindingsJs.FileGenerationOptions(); //TODO (comment) this was a pointer
+            fileGenerationOpts[fileId].setFileTypeToGenerate(bindingsJs.FileGenerationOptions.FileTypeToGenerate.FILE);
+        }
+
+        let args = constructors.newGetGenerateFilesArgs(serviceId, fileGenerationOpts);
+        let request: grpc.requestCallback<bindingsJs.GenerateFilesResponse>;
+
+        let resp, err = this.client.generateFiles(args, request);
```

Comment:
This request is suppose to ressemble the callback feature. However, Kevin showed me that I should ideally use the `await` command in order to do a callback. I played around with this for quite a bit, but I need to see if I could use `await` command along with this `generateFiles` call (I'm having my doubts if I actually need this method since the return type is not `ExecCommandResponse`).